HIP/hip-1037.md

```diff
@@ -0,0 +1,95 @@
+possible to produce a compatible independent implementation of the Hedera
+consensus node software without extensive reference to the existing source
+code. Second, developers are often unclear how best to interact with the
+Hedera API, and independent SDK developers, in particular, may struggle to
+correctly implement these API calls.
+As part of expanding the Hedera ecosystem we MUST reduce the difficulty and
+hurdles for independent developers when building systems that interact with
+the Hedera network. Clear and accurate API specifications are one element
+required to accomplish this goal.
+
+## Rationale
+This HIP improves the developer experience and enhances decentralization by
+making it easier to design distributed applications or alternative
+implementations of core network systems.
+The use of a markdown format that is automatically generated from the
+functional protocol buffer definition files encourages more frequent updates
+that are more closely tied to the functional updates to the API documents, and
+reduces the inevitable "drift" between functional API and API specification.
+
+## User stories
+- As a dApp developer I want to have clear and accurate specification for all
+  Hedera APIs so that I can confidently design my application to call those
+  APIs.
+- As an independent implementor I want to have clear and accurate specification
+  for all Hedera APIs so that I can design an independent, correct, and
+  compatible implementation of Hedera network nodes, including a consensus node.
+- As an Hedera software contributor I want to have clear and accurate
+  specification for all Hedera APIs so that I can confidently implement changes
+  to Hedera-led software.
+
+## Specification
+This HIP makes sweeping changes to specification text for the existing API
+documents. These changes affect over 20,000 lines of text across almost 150
+files, which is excessive for a document such as this HIP. The full
+specification, therefore, is only linked here for brevity and clarity.<br/>
+The specification changes intended for this HIP are detailed in github as a
+[pull request](https://github.com/hashgraph/hedera-protobufs/pull/388)
+for the hedera-protobufs repository.
+
+## Backwards Compatibility
+This HIP does not change any functional characteristic, and documents the
+state of the API as-is.  This does not introduce any change in compatibility.
+
+## Security Implications
+This HIP updates and clarifies specification and expectation for the Hedera
+API. This does not materially impact the security of the system.
+
+## How to Teach This
+The Hedera API is a critical component of the design and interaction for the
+Hedera ecosystem. Improvements to the formality and completeness of the API
+specification help everyone involved to develop distributed applications
+more quickly and with greater confidence.
+
+## Reference Implementation
+Not applicable
+
+## Rejected Ideas
+Not applicable
+
+## Open Issues
+None.
+
+## References
```

Comment:
Can point to [`proto-gen-doc`'s](https://github.com/pseudomuto/protoc-gen-doc/blob/master/README.md) readme right here.

Reply:
Good idea, added using the main-page sublink.